File: apps/console-web/src/types/event.ts
export interface XDREvent {
  id: string
  event_id: string
  tenant_id: string
  source: string
  event_type: string
  severity: string
  asset_id?: string
  device_id?: string
  identity_id?: string
  observed_at: string
  raw?: Record<string, unknown>
  normalized?: Record<string, unknown>
  risk_score?: number
  risk_factors?: string[]
  collector?: string
  pipeline?: string
  created_at: string
}

export interface Playbook {
  id: string
  name: string
  description?: string
  version: string
  trigger_type?: string
  is_active: boolean
  created_at: string
  updated_at: string
}

File: services/event-normalizer/internal/models/event.go
package models

import "time"

type XDREvent struct {
	EventID       string                 `json:"event_id"`
	TenantID      string                 `json:"tenant_id"`
	Source        string                 `json:"source"`
	EventType     string                 `json:"event_type"`
	Severity      string                 `json:"severity"`
	AssetID       string                 `json:"asset_id,omitempty"`
	DeviceID      string                 `json:"device_id,omitempty"`
	IdentityID    string                 `json:"identity_id,omitempty"`
	ObservedAt    time.Time              `json:"observed_at"`
	Raw           map[string]interface{} `json:"raw,omitempty"`
	Normalized    map[string]interface{} `json:"normalized,omitempty"`
	Risk          *RiskScore             `json:"risk,omitempty"`
	Trace         *TraceInfo             `json:"trace,omitempty"`
	MITRE         *MITREInfo             `json:"mitre,omitempty"`
}

type RiskScore struct {
	Score   float64  `json:"score"`
	Factors []string `json:"factors"`
}

type TraceInfo struct {
	Collector     string `json:"collector"`
	Pipeline      string `json:"pipeline"`
	SchemaVersion string `json:"schema_version"`
}

type MITREInfo struct {
	Tactics    []string `json:"tactics"`
	Techniques []string `json:"techniques"`
}
